uber/site_sections/registration.py

```diff
@@ -467,12 +467,7 @@ def take_payment(self, session, payment_id, stripeToken):
             else:
                 attendee.paid = HAS_PAID
                 attendee.amount_paid = attendee.total_cost
-
-                # HACK! need to fix this to save correctly.
-                if attendee.registered is None:
-                    attendee.registered = datetime.now()
-
-                session.add(attendee)
+                session.merge(attendee)
```

Comment:
I've explained the difference between ``add`` and ``merge`` elsewhere, but the tl;dr is that ``merge`` is for when you have an existing object which you want to attach to a session but represents something already in the database.